From 4d5f99a0069f076ec795e433f956822153bda22c Mon Sep 17 00:00:00 2001
From: Ranjani Vaidyanathan <Ranjani.Vaidyanathan@nxp.com>
Date: Thu, 28 Sep 2017 13:48:23 -0500
Subject: [PATCH 2647/5242] MLK16557 soc:imx8qm/imx8qx - Resources can request
 low power idle mode only when runtime-pm is
 enabled.

commit  806424bf607bb53fff717569e44f2513e46128b6 from
https://source.codeaurora.org/external/imx/linux-imx.git

Some drivers use runtime PM callbacks during suspend/resume also and this
in turn results in SCFW calls requesting the resource to enter
low power idle instead OFF state.
This patch fixes this issue by ensuring that low power IDLE request is only
valid when runtime PM is enabled. Runtime PM is disabled when the system is
entering suspend state.

BuildInfo: SCFW 7a725203, IMX-MKIMAGE ee6adff0, ATF 0

Signed-off-by: Ranjani Vaidyanathan <Ranjani.Vaidyanathan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/pm-domains.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/soc/imx/pm-domains.c b/drivers/soc/imx/pm-domains.c
index 5626315..a99883a 100644
--- a/drivers/soc/imx/pm-domains.c
+++ b/drivers/soc/imx/pm-domains.c
@@ -133,7 +133,8 @@ static int imx8_pd_dev_stop(struct device *dev)
 	struct imx8_pm_domain *pd;
 
 	pd = container_of(genpd, struct imx8_pm_domain, pd);
-	pd->runtime_idle_active = true;
+	if (pm_runtime_enabled(dev))
+		pd->runtime_idle_active = true;
 	return 0;
 }
 
@@ -143,7 +144,8 @@ static int imx8_pm_runtime_idle(struct device *dev)
 	struct imx8_pm_domain *pd;
 
 	pd = container_of(genpd, struct imx8_pm_domain, pd);
-	pd->runtime_idle_active = true;
+	if (pm_runtime_enabled(dev))
+		pd->runtime_idle_active = true;
 	return pm_runtime_autosuspend(dev);
 }
 
-- 
1.7.9.5

